commerce/domain/shipping/splitshipment/SplitShipmentSample.cls

```diff
@@ -0,0 +1,158 @@
+/**
+ * @description Custom Split Shipment Service sample.
+ */
+global class SplitShipmentSample extends CartExtension.SplitShipmentService {
+
+    public virtual override void arrangeItems(CartExtension.ItemArrangementRequest request) {
+        CartExtension.Cart cart = request.getCart();
+        Iterator<CartExtension.ItemArrange> itemArrangeIterator = request.getItemArrangeList().iterator();
+
+        // Get all productIds from the ItemArrange list
+        List<Id> productIds = new List<Id>();
+        while (itemArrangeIterator.hasNext()) {
+            CartExtension.ItemArrange itemArrange = itemArrangeIterator.next();
+            productIds.add(itemArrange.getProductId());
+        }
+
+        // Get conveyability info for each product
+        List<Product2> products = [
+            SELECT
+                Id
+                , IsConveyable__c
+            FROM Product2
+            WHERE Id IN :productIds
+        ];
+        Iterator<Product2> productsIterator = products.iterator();
+
+        Map<Id,Boolean> productConveyability = new Map<Id,Boolean>();
+        while (productsIterator.hasNext()) {
+            Product2 product = productsIterator.next();
+            productConveyability.put(product.Id, product.IsConveyable__c);
+        }
+
+        // Arrange items by conveyability
+        itemArrangeIterator = request.getItemArrangeList().iterator();
+        while (itemArrangeIterator.hasNext()) {
+            CartExtension.ItemArrange itemArrange = itemArrangeIterator.next();
+
+            // If the item is non-conveyable, then move it to a non-conveyable CartDeliveryGroup
+            if (productConveyability.get(itemArrange.getProductId()) != true) {
```

Comment:
just checking if Apex handles it without issues. The `!=` true comparison would not fail if there is no `productId` in the `Map` and `null` is returned?

Reply:
Yes Apex handles this correctly. I ran into a scenario while testing the script in an org where the `productConveyability` map was not getting populated due to issues with the SOQL query that fetches product data. This condition works as expected when `null` is returned from the map.